dbms/src/TiDB/Decode/JsonScanner.cpp

```diff
@@ -0,0 +1,516 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <TiDB/Decode/JsonScanner.h>
+
+namespace DB
+{
+bool JsonScanIsSpace(char c)
+{
+    return (static_cast<unsigned char>(c) <= static_cast<unsigned char>(' '))
+        && (c == ' ' || c == '\t' || c == '\r' || c == '\n');
+}
+
+bool checkJsonValid(const char * data, size_t length)
```

Comment:
how about using `SIMDJson` directly?

Reply:
Not sure if SIMDJson behaves exactly the same with go-lib.

Reply:
SIMDJson can take advantage of AVX2 and AVX512 to improve performance

Reply:
Yeah, it's not common case to do "valid check" here after introduce need_check flag. Thus I prefer use go-lib one currently, and I'll add TODOs to use simdJson in future.

Reply:
ok.